### PR TITLE
Revert "[ci.yaml] Test web engine on Mac 12"

### PR DESCRIPTION
**Pull request by keyonghan**
Reverts flutter/engine#32337

The Mac Web Engine is timing out: https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20Web%20Engine/9556/overview

Context: https://github.com/flutter/flutter/issues/97303

### CODE REVIEW

**Review by CaseyHillers — APPROVED** [review]

**Comment by CaseyHillers** [comment]
@keyonghan before landing, did we wipe the caches of the prod Mac 10 bots? I still lack ssh access

**Comment by jmagman** [comment]
They are timing out because of https://github.com/flutter/flutter/issues/99885

**Comment by keyonghan** [comment]
Manually land to unblock the tree.

**Pull request merged**
